### ISSUE TITLE
Feature read __sample_limit__ from labels

### ISSUE THREAD

**Issue by zoglam** [VictoriaMetrics/VictoriaMetrics#6665]
### Is your feature request related to a problem? Please describe

Currently, there is no functionality to override the `sampleLimit` parameter value from the job configuration if a label `__sample_limit__` is attached to the targets. (like `__series_limit__` which is exists)

### Describe the solution you'd like

Introduce a new feature that enables the system to overwrite the `sampleLimit` parameter value in the job configuration when a label `__sample_limit__` is specified for target.

This enhancement would provide flexibility and allow users to customize the sampleLimit parameter based on specific target requirements.

### Additional information

/lib/promscrape/config.go (https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/lib/promscrape/config.go#L1182-L1189)
```go
// ...
sampleLimit := swc.sampleLimit
if s := labels.Get("__sample_limit__"); len(s) > 0 {
  n, err := strconv.Atoi(s)
  if err != nil {
    return nil, fmt.Errorf("cannot parse __sample_limit__=%q: %w", s, err)
  }
  sampleLimit = n
}

// ...

sw := &ScrapeWork{
// ...
  SampleLimit:          sampleLimit,
// ...
}
```

**Comment by jiekun** [comment]
> This enhancement would provide flexibility and allow users to customize the sampleLimit parameter based on specific target requirements.

Hi. CMIIW, it's like using `sampleLimit` as a global config (for a job) and adjusting some targets by modifying their `__sample_limit__` label. So the use case is: wide range sample limit control + special control for some targets?

**Comment by zoglam** [comment]
idk how to assign PR correctly. But here's a mention
https://github.com/VictoriaMetrics/VictoriaMetrics/pull/6666

**Comment by zoglam** [comment]
> > This enhancement would provide flexibility and allow users to customize the sampleLimit parameter based on specific target requirements.
> 
> Hi. CMIIW, it's like using `sampleLimit` as a global config (for a job) and adjusting some targets by modifying their `__sample_limit__` label. So the use case is: wide range sample limit control + special control for some targets?

yeap

**Comment by jiekun** [comment]
Hi big thanks for the pull request. I will discuss with maintainers later. 
Please notice that we may not accept all new features from the community based on [VictoriaMetrics development goals](https://docs.victoriametrics.com/goals/). I will comment here once we have update for you.

**Comment by hagen1778** [comment]
This feature was included into [v1.103.0](https://docs.victoriametrics.com/changelog/#v11030)
